### ISSUE TITLE
GPU compute_ready flag is decoupled from actual channel handoff state

### ISSUE THREAD

**Issue by johnjezl** [SLM-OS/SLM-Operating-System#702]
## Summary

`slm_gpu_get_info()` in `kernel/src/slm_ffi.c:344-353` hardcodes `info->compute_ready = 1` on Jetson via `#ifdef PLATFORM_JETSON_ORIN_NANO`, decoupled from any runtime check of whether the GA10B channel handoff actually succeeded.

```c
#ifdef PLATFORM_JETSON_ORIN_NANO
    info->compute_ready = 1;
#else
    info->compute_ready = 0;
#endif
```

## Why this matters

The runtime side trusts this flag in `runtime/src/inference/gpu.rs::GpuCapabilities::detect()` — `status = ComputeReady` whenever `info.compute_ready != 0`. `mnist_gpu_fastpath_eligible()` in `runtime/src/inference/engine.rs` then uses `has_compute()` as one of four gates to decide whether to dispatch GPU MNIST.

Because the flag is a static `#ifdef`, the eligibility gate passes even when:

- Default `slmos-kexec` (suspend mode) was used — runtime PM drained Linux's GPU state and the helper-staged handoff isn't present
- GSP firmware is missing (boot log shows `[GSP] firmware missing: gsp / bootloader / booter_load / booter_unload` + `[WARN] GSP init incomplete (rc=-1) — GPU compute unavailable`)
- v6 channel-inherit failed and v7 fallback didn't successfully prepare the QMD pool

This is the suspected root cause behind #692: the user-visible \"GPU MNIST\" path returns logits bit-identical to CPU because dispatch is silently CPU-only, but the eligibility gate still says GPU is ready.

## Proposed fix

1. Make `compute_ready` reflect actual handoff state — set it from `g_handoff.version != 0` and `g_handoff.pipeline_n_ops > 0` (or the v7 equivalent), not just platform identity.
2. Extend `gpu use status` to surface the handoff diagnostics (`g_handoff.version`, `g_handoff.pipeline_n_ops`, last dispatch rc) so the operator can see why the gate is closed without rebuilding with diagnostics.
3. When eligibility fails at the dispatch boundary, log unconditionally to UART (not behind Info-level filtering) so the failure mode is visible without re-instrumentation.

## Related

- #692 — GPU MNIST returns CPU-identical logits (this is the suspected mechanism)
- PR #694 — v7+HMMA structural fix (does not address this audit; orthogonal)

## Acceptance criteria

- `compute_ready=0` on a default suspend-kexec boot where helper-staged handoff isn't present
- `gpu use status` shows handoff version and op count
- Eligibility-gate failures are visible on UART without Info-level filtering
